openml/runs/functions.py

```diff
@@ -391,26 +395,103 @@ def _run_task_get_arffcontent(
+                )
+
                 (
-                    arff_datacontent_fold,
-                    arff_tracecontent_fold,
+                    pred_y,
+                    proba_y,
                     user_defined_measures_fold,
-                    model_fold,
+                    trace,
                 ) = extension._run_model_on_fold(
                     model=model,
                     task=task,
+                    X_train=train_x,
+                    y_train=train_y,
                     rep_no=rep_no,
                     fold_no=fold_no,
-                    sample_no=sample_no,
-                    add_local_measures=add_local_measures,
+                    X_test=test_x,
+                    classes=classes,
                 )
 
+                arff_datacontent_fold = []  # type: List[List]
+                if trace is not None:
+                    traces.append(trace)
+
+                # add client-side calculated metrics. These is used on the server as
+                # consistency check, only useful for supervised tasks
+                def _calculate_local_measure(sklearn_fn, openml_name):
+                    user_defined_measures_fold[openml_name] = sklearn_fn(test_y, pred_y)
+
+                if isinstance(task, (OpenMLClassificationTask, OpenMLLearningCurveTask)):
+
+                    for i in range(0, len(test_indices)):
+
+                        arff_line = [rep_no, fold_no, sample_no, i]  # type: List[Any]
+                        for j, class_label in enumerate(task.class_labels):
+                            arff_line.append(proba_y[i][j])
+
+                        arff_line.append(task.class_labels[pred_y[i]])
+                        arff_line.append(task.class_labels[test_y[i]])
+
+                        arff_datacontent.append(arff_line)
+
+                    if add_local_measures:
+                        _calculate_local_measure(
+                            sklearn.metrics.accuracy_score,
+                            'predictive_accuracy',
+                        )
+
+                elif isinstance(task, OpenMLRegressionTask):
+
+                    for i in range(0, len(test_indices)):
+                        arff_line = [rep_no, fold_no, test_indices[i], pred_y[i], test_y[i]]
+                        arff_datacontent.append(arff_line)
+
+                    if add_local_measures:
+                        _calculate_local_measure(
+                            sklearn.metrics.mean_absolute_error,
+                            'mean_absolute_error',
+                        )
+
+                elif isinstance(task, OpenMLClusteringTask):
+                    for i in range(0, len(test_indices)):
+                        arff_line = [test_indices[i], pred_y[i]]  # row_id, cluster ID
+                        arff_datacontent.append(arff_line)
+
+                else:
+                    raise TypeError(type(task))
+
                 arff_datacontent.extend(arff_datacontent_fold)
```

Comment:
What is the use `arff_datacontent_fold`? It is defined in line 446 but I don't see it being used anywhere else. Is this an error? Or is it legacy? If it is legacy, I would prefer `arff_datacontent.extend([])  # legacy` or something equivalent.

Reply:
It's not used any more so I removed it.